src/providers/commandsProvider.ts

```diff
@@ -94,25 +99,21 @@ function execCommand(currentRedPath: string, command: string, fileUri?: vscode.U
         filePath = fileUri.fsPath;
     }
 
-    terminal = terminal ? terminal : vscode.window.createTerminal(`Red`);
+    let outputFilename = path.join(path.parse(filePath).dir, path.parse(filePath).name);
     
     switch(command) {
         case Commands.Red_Interpret: {
             if (redTool) {
-                text = `${currentRedPath} --cli "${filePath}"`;
+                text = `${redExecutable} --cli "${filePath}"`;
             } else {
-                text = `${currentRedPath} "${filePath}"`;
+                text = `${redExecutable} "${filePath}"`;
             }
         } break;
         case Commands.Red_Compile: {
-            buildDir = redSettings.buildDir ||
```

Comment:
Why remove buildDir?

Reply:
Earlier buildDir was being used to `cd` into the directory that contains the source code.
But then the issue was that if red executable is in a directory with spaces, the command executed would be something like this, which is wrongs because the space breaks up the command.
```
C:\red space\red.exe -c "C:\Users\abc\desktop\a.red"
```
I tried to add quotes to the red executable path so that the command would become
```
"C:\red space\red.exe" -c "C:\Users\abc\desktop\a.red"
```
but that is not portable even on windows, for eg. it works on `cmd` but not on `powershell`.

So what I did was to `cd` into the directory containing `red.exe`. And after that pass the full path to the source code as well as the output file name. 
Therefore the earlier terminal command now changes to -
```
cd "C:\red space\"
red.exe -o "C:\Users\abc\desktop\a" -c "C:\Users\abc\desktop\a.red"
```

Since the output IMO should be generated in the same directory as the source code (atleast until a build/package manager arises), therefore AFAICT there is no need for having a `buildDir` right now.

Reply:
Yes. `cd` into the directory containing `red.exe` is better. It would be better to keep `red.buildDir`. Just check if it is set or not. If it's set, just output to the buildDir.

Reply:
Re-added buildDir.

> Just check if it is set or not.
  
I don't think that would be needed since if `buildDir` is not set, it automatically sets it to `filePath`